portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -51,7 +54,32 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     Gson gson = new Gson();
     response.getWriter().println(gson.toJson(comments));
   }
-  
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String text = request.getParameter("text-input");
+    long timestamp = System.currentTimeMillis();
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("text", text);
+    commentEntity.setProperty("timestamp", timestamp);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    response.sendRedirect("/comments.html");
+  }
+
+  @Override
+  public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String idString = request.getPathInfo().substring(1);
+    long id = Long.parseLong(idString);
```

Comment:
One error case to consider, what happens if the string is not an  integer? Also what happens if the id is not found?

Reply:
It's definitely important to address these cases. I have added the edge cases. Done.